src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```
@@ -0,0 +1,110 @@
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(parseManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to get all manufacturers ", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String query = "UPDATE manufacturers SET name = ?,"
+                + " country = ? WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.setLong(3, manufacturer.getId());
+            if (preparedStatement.executeUpdate() > 0) {
+                return manufacturer;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to update manufacturer " + manufacturer, e);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturers SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to delete manufacturer with id: " + id, e);
+        }
+    }
+
+    private Manufacturer parseManufacturer(ResultSet resultSet) {
+        try {
```

we can just rethrow this SQL exception into the method signature because it is already surrounded by the try-catch